### PR TITLE
[SAGE-497] Label - align to spec

### PR DESCRIPTION
**Pull request by QuintonJason**
Closes [SAGE-497](https://github.com/Kajabi/sage-lib/pull/497)
Dependent on [SAGE-554](https://github.com/Kajabi/sage-lib/pull/554)
## Description

- [x] align label to spec
    - [x] update interactive border-radius
    - [x] reduce label text font-weight

## Screenshots

|  Before  |  After  |
|--------|--------|
|Screen Shot 2021-06-21 at 8 51 06 AM|Screen Shot 2021-06-21 at 8 50 26 AM|

## Testing in `sage-lib`

Visit the label rails and react page 

## Testing in `kajabi-products`

1. (LOW) Label component adjusted to latest specs. Label changes will occur as a result in kajabi-products but no adverse effects expected.

### CODE REVIEW

File: packages/sage-assets/lib/stylesheets/components/_button.scss [hunk 1]
```diff
@@ -174,22 +174,23 @@ $-btn-interactive-label-icon-size: rem(24px);
     right: 0;
     width: $-btn-interactive-label-icon-size;
     margin: auto 0;
-    border-radius: 0 sage-border(radius) sage-border(radius) 0;
+    border-radius: 0 sage-border(radius-x-large) sage-border(radius-x-large) 0;
+
+    &::before {
+      margin-top: 1px; /* needed as sage-font-size(sm) is equates 13px and not 14px. This 1px is accounting for that 1px */
```
Comment (monkeypox8) [inline packages/sage-assets/lib/stylesheets/components/_button.scss:180]: I think we'll want to handle this with a `transform`, since the margin adds to the secondary button height on hover:

**Review by monkeypox8 — COMMENTED** [review]
Looking fantastic! Just to confirm, will the focus outline states be handled in a separate button/interaction PR?

File: packages/sage-assets/lib/stylesheets/components/_button.scss (hunk 1 above)
Comment (QuintonJason) [reply packages/sage-assets/lib/stylesheets/components/_button.scss]: Great catch! Pushed the fix

**Review by monkeypox8 — APPROVED** [review]
👍🏼 LGTM!

Firefox seems to be rendering the alignment in its own way from Chrome/Safari, so the added transform/margin ends up pushing the button an additional 1px off the vertical line. I don't think it's a major concern, though:

**Review by philschanely — APPROVED** [review]
Looks and works well. 🙌  One non-blocker is this older interactive—effect which I imagine will go away soon given your fabulous new interactive effects PR—does look a little odd with the new pill shape:

Screen Shot 2021-06-24 at 10 27 05 AM

Definitely an optional fix but wanted to raise awareness just in case it was of interest to fix now.

**Review by philschanely — APPROVED** [review]
Looking good, @QuintonJason . 

These may deserve a follow-up or two, but I noted the following:

1. To match the specs closer it appears we need to revise the secondary button hover settings a bit:

    Spec:

    Screen Shot 2021-07-19 at 12 01 42 PM

    As built to date:
    Screen Shot 2021-07-19 at 12 10 55 PM

2. There are also some aspects of how the colors and the mixture of sage btn within this component for the secondary button feel potentially leaky that might be nice revisit. See color settings in loop in `_label.scss` that override those in `_button.scss`. I wonder if we indeed need to use `SageButton` there or if the tradeoff is in our favor to use a custom styled HTML `button` instead?

Bottom line is nothing blocking this PR but some ideas about potential follow up. If you agree, I'm happy to file them in an issue for later. Thoughts?

**Comment by QuintonJason** [comment]
> Looking good, @QuintonJason .
> 
> These may deserve a follow-up or two, but I noted the following:
> 
> 1. To match the specs closer it appears we need to revise the secondary button hover settings a bit:
>    Spec:
>     Screen Shot 2021-07-19 at 12 01 42 PM
>    As built to date:
>    Screen Shot 2021-07-19 at 12 10 55 PM
> 2. There are also some aspects of how the colors and the mixture of sage btn within this component for the secondary button feel potentially leaky that might be nice revisit. See color settings in loop in `_label.scss` that override those in `_button.scss`. I wonder if we indeed need to use `SageButton` there or if the tradeoff is in our favor to use a custom styled HTML `button` instead?
> 
> Bottom line is nothing blocking this PR but some ideas about potential follow up. If you agree, I'm happy to file them in an issue for later. Thoughts?

@philschanely Great callout! Made one commit to update hover colors, but we can revisit the need for the `SageButton` in a followup

**Comment by QuintonJason** [comment]
@philschanely https://github.com/Kajabi/sage-lib/issues/640
